test/recipes/15-test_genrsa.t

```diff
@@ -18,9 +18,34 @@ setup("test_genrsa");
 
 plan tests => 5;
 
+# We want to know that an absurdly small number of bits isn't support
 is(run(app([ 'openssl', 'genrsa', '-3', '-out', 'genrsatest.pem', '8'])), 0, "genrsa -3 8");
-ok(run(app([ 'openssl', 'genrsa', '-3', '-out', 'genrsatest.pem', '16'])), "genrsa -3 16");
-ok(run(app([ 'openssl', 'rsa', '-check', '-in', 'genrsatest.pem', '-noout'])), "rsa -check");
-ok(run(app([ 'openssl', 'genrsa', '-f4', '-out', 'genrsatest.pem', '16'])), "genrsa -f4 16");
-ok(run(app([ 'openssl', 'rsa', '-check', '-in', 'genrsatest.pem', '-noout'])), "rsa -check");
+
+# Depending on the shared library, we might have different lower limits.
+# Let's find it!
+note "Looking for lowest amount of bits";
+my $bad = 3;                    # Number of bits
+my $good = 11;                  # Number of bits
```

Comment:
No, they are not. I mean these are not amount of bits, but binary logarithm of amount of bits, or number of bits in number holding number of [good] bits.

However! I'd rather suggest to simply take the recipe from master to 1.1.0. It's really no point in testing such low numbers.

Reply:
The script from master will fail because of this line:

```
is(run(app([ 'openssl', 'genrsa', '-3', '-out', 'genrsatest.pem', '256'])), 0, "genrsa -3 256");
```

Of course, that particular line can be left with `'8'` rather than `'256'`.  It *is* designed to check that we do have a lower bound, even though absurd.

I'm also thinking that some time in the future, we'll end up with the exact same problem again, just with higher numbers.  Someone will get tired of the repeating scenario...

Re comment on number of bits, of course you're right.

Reply:
As for future, this won't cross 1.1.x boundary, so it will be just limited number of deviations. And if one puts [that much] value into checking the lower boundary, it can be the only difference between the branches. I mean one always takes recipe from master and adjusts down only the first test for lower bound.

Concern with iterative search is that algorithm is hard to follow... And formally speaking it might need adjustment in new version anyway. Because choice of constants is formally version specific. I mean there is a cap...

Reply:
> there is a cap

For example if you set minimum to any value over 2048 it will fail to find right value. I'm not saying that we should set it to 2048, only that suggested algorithm has limitations (that one has to keep in mind and take into consideration). I'd say that examining openssl_versions output (introduced in #5753) would be more appropriate. Especially with commentary that says that limit modifications has to be backported to earlier minor versions.

Reply:
Either way, there will be something to maintain at some point, it seems.  I fail to understand why one is superior to the other...

Reply:
> I fail to understand why one is superior to the other...

As already said, suggested algorithm is hard to follow, and consequently hard to maintain. As practical example, if one wants to increase the cap beyond 2048, how many minutes will it take for somebody else to figure out how to do it and at the same time ensure that it still work in 1.1.0 context. Compare it to adding *one* number, minimum supported, to lookup table with version as "index".